cmd/dockerd/daemon.go

```diff
@@ -647,17 +675,20 @@ func checkTLSAuthOK(c *config.Config) bool {
 	return true
 }
 
-func loadListeners(cli *DaemonCli, tlsConfig *tls.Config) ([]string, error) {
-	if len(cli.Config.Hosts) == 0 {
-		return nil, errors.New("no hosts configured")
+func loadListeners(cfg *config.Config, tlsConfig *tls.Config) ([]net.Listener, []string, error) {
```

Comment:
❤️ we really need to make a pass at some of these. Too many places where we just pass everything (mostly out of convenience)